### PR TITLE
add gas price to artifact

### PR DESCRIPTION
**Pull request by Jim8y**
this pr adds the gas price to the artifact, such that we can easily check the optimization detail in git diff.

### CODE REVIEW

File: tests/Neo.Compiler.CSharp.UnitTests/TestingArtifacts/Contract_Recursion.cs [hunk 1]
```diff
@@ -26,26 +26,26 @@ public abstract class Contract_Recursion(Neo.SmartContract.Testing.SmartContract
...
+    /// 00 : OpCode.INITSLOT 0001	[64 datoshi]
+    /// 03 : OpCode.LDARG0	[2 datoshi]
+    /// 04 : OpCode.PUSH0	[1 datoshi]
+    /// 05 : OpCode.EQUAL	[32 datoshi]
+    /// 06 : OpCode.JMPIFNOT 04	[2 datoshi]
+    /// 08 : OpCode.PUSHT	[1 datoshi]
+    /// 09 : OpCode.RET	[0 datoshi]
+    /// 0A : OpCode.LDARG0	[2 datoshi]
+    /// 0B : OpCode.PUSH0	[1 datoshi]
+    /// 0C : OpCode.LT	[8 datoshi]
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/TestingArtifacts/Contract_Recursion.cs:38]: @Jim8y Why some of them are not regenerated?

Comment (Jim8y) [reply tests/Neo.Compiler.CSharp.UnitTests/TestingArtifacts/Contract_Recursion.cs:38]: I am not sure what you mean, looks like all have being updated.

File: tests/Neo.Compiler.CSharp.UnitTests/TestingArtifacts/Contract_Assignment.cs [hunk 2]
```diff
@@ -55,36 +55,36 @@ public abstract class Contract_Assignment(Neo.SmartContract.Testing.SmartContrac
...
+    /// 18 : OpCode.LDLOC0	[2 datoshi]
+    /// 19 : OpCode.JMP 05	[2 datoshi]
+    /// 1B : OpCode.PUSH2	[1 datoshi]
+    /// 1C : OpCode.DUP	[2 datoshi]
+    /// 1D : OpCode.STLOC0	[2 datoshi]
+    /// 1E : OpCode.DROP	[2 datoshi]
+    /// 1F : OpCode.LDLOC0	[2 datoshi]
+    /// 20 : OpCode.PUSH1	[1 datoshi]
+    /// 21 : OpCode.EQUAL	[32 datoshi]
+    /// 22 : OpCode.ASSERT	[1 datoshi]
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/TestingArtifacts/Contract_Assignment.cs:86]: @Jim8y Why some use space and other tabs?

**Comment by Jim8y** [comment]
@shargon fixed. the reason of that is because the only difference between old artifact and new artifact is now many space are there, while the comparison actually will ignore space.... thus it will not be updated.

File: tests/Neo.Compiler.CSharp.UnitTests/TestingArtifacts/Contract2.cs [hunk 3]
```diff
@@ -28,7 +28,7 @@ public abstract class Contract2(Neo.SmartContract.Testing.SmartContractInitializ
     /// Script: VwECDAECAwTbMHBoEs5A
     /// 00 : OpCode.INITSLOT 0102 [64 datoshi]
     /// 03 : OpCode.PUSHDATA1 01020304 [8 datoshi]
-    /// 09 : OpCode.CONVERT 30 [8192 datoshi]
+    /// 09 : OpCode.CONVERT (Buffer) [8192 datoshi]
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/TestingArtifacts/Contract2.cs:31]: 0x30=Buffer?

Comment (Jim8y) [reply tests/Neo.Compiler.CSharp.UnitTests/TestingArtifacts/Contract2.cs]: done

File: src/Neo.Disassembler.CSharp/Disassembler.cs [hunk 4]
```diff
@@ -60,16 +61,47 @@ private static (int start, int end) GetMethodStartEndAddress(string name, JToken
...
+    public static string InstructionToString(this Instruction instruction, bool addPrice = true)
     {
         var opcode = instruction.OpCode.ToString();
         var operand = instruction.Operand;
 
-        if (operand.IsEmpty || operand.Length == 0)
+        var addprice = 0L;
+        string ret;
+
+        if (operand.IsEmpty)
```
Comment (shargon) [inline src/Neo.Disassembler.CSharp/Disassembler.cs:72]:
```suggestion
        if (operand.IsEmpty || operand.Length == 0)
```

**Review by shargon — APPROVED** [review]

**Pull request merged**